query-engine/connectors/mongodb-query-connector/src/error.rs

```diff
@@ -162,3 +162,19 @@ impl From<mongodb::bson::oid::Error> for MongoError {
         MongoError::MalformedObjectId(format!("{}", err))
     }
 }
+
+impl From<mongodb::bson::extjson::de::Error> for MongoError {
+    fn from(err: mongodb::bson::extjson::de::Error) -> Self {
+        match err {
+            mongodb::bson::extjson::de::Error::InvalidObjectId(oid_err) => oid_err.into(),
+            mongodb::bson::extjson::de::Error::DeserializationError { message: _ } => MongoError::ConversionError {
+                from: "JSON".to_string(),
+                to: "BSON".to_string(),
+            },
+            // Needed because `extjson::de::Error` is flagged as #[non_exhaustive]
+            _ => MongoError::UnhandledError(
```

Comment:
You can catch the generic error with `err => ...` and simply dump it in the error message if it supports either debug or display.